server/cloudformation/cloudformation.go

```diff
@@ -224,3 +228,67 @@ func (c *CustomResourceProvisioner) provision(ctx context.Context, m Message, re
 
 	return p.Provision(ctx, req)
 }
+
+type properties interface {
+	ReplacementHash() (uint64, error)
+}
+
+// provisioner provides convenience over the customresources.Provisioner
+// interface.
+type provisioner struct {
+	properties func() properties
+
+	Create func(context.Context, customresources.Request) (string, interface{}, error)
+	Update func(context.Context, customresources.Request) (interface{}, error)
+	Delete func(context.Context, customresources.Request) error
+}
+
+func (p *provisioner) Properties() interface{} {
+	return p.properties()
+}
+
+func (p *provisioner) Provision(ctx context.Context, req customresources.Request) (string, interface{}, error) {
```

Comment:
This basically just abstracts the commonality between all of the resource provisioners. Only the new `ECSEnvironment` and `ECSTaskDefinition` resources are using this so far. Eventually, we should move the `ECSService` and `InstancePort` resources over to this as well.